### PR TITLE
LPS-159063 Allow sort by External Reference Code Field

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-159063

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
This PR will fail on SF until this PR #300  be merge

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/entity/v1_0/ObjectEntryEntityModel.java [hunk 1]
```diff
@@ -58,7 +68,9 @@ public ObjectEntryEntityModel(List<ObjectField> objectFields) {
 				"dateModified", locale -> "modifiedDate",
 				locale -> "modifiedDate")
 		).put(
-			"externalReferenceCode", new StringEntityField("id", locale -> "id")
+			"externalReferenceCode",
+			new StringEntityField(
+				externalReferenceCode, locale -> finalExternalReferenceCode)
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/entity/v1_0/ObjectEntryEntityModel.java:73]: Hi @p-albuquerque,

Here I believe we can do something like this:

````
.put(
	"externalReferenceCode",
	() -> {
	   if (GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-158821"))) {
			return new StringEntityField(
				"externalReferenceCode", locale -> "externalReferenceCode");
		}
	
	   return new StringEntityField("id", locale -> "id");
	}
)

````

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 2]
```diff
@@ -1200,7 +1200,7 @@ public void testGetValuesList() throws Exception {
 		Assert.assertEquals("John", values.get("firstName"));
 		Assert.assertEquals(
 			"listTypeEntryKey3", values.get("listTypeEntryKeyRequired"));
-		Assert.assertEquals(values.toString(), 24, values.size());
+		Assert.assertEquals(values.toString(), 25, values.size());
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:1203]: Hi @p-albuquerque,

What does this number (25) means?

Let's avoid "magic numbers" by trying to extract this number to a final variable with a name that makes easy the comprehension, talk to me when you desired if you need support to do it.

Thanks!

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/frontend/data/set/view/table/ObjectEntriesTableFDSView.java [hunk 3]
```diff
@@ -206,12 +214,15 @@ else if (Objects.equals(fieldName, "createDate")) {
 				null, null, "Date", fdsTableSchemaBuilder, "dateCreated",
 				fieldLabel, true);
 		}
-		else if (Objects.equals(fieldName, "externalReferenceCode") ||
-				 Objects.equals(fieldName, "id")) {
-
+		else if (Objects.equals(fieldName, "externalReferenceCode")) {
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/frontend/data/set/view/table/ObjectEntriesTableFDSView.java:217]: Hi @p-albuquerque,

Can we do something like this?

````
else if (Objects.equals(fieldName, "externalReferenceCode") &&
	 GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-158821"))) {

	 _addFDSTableSchemaField(
				null, "actionLink", null, fdsTableSchemaBuilder,
				"externalReferenceCode", fieldLabel, true);
}
else if (Objects.equals(fieldName, "id")) {
			_addFDSTableSchemaField(
				null, "actionLink", null, fdsTableSchemaBuilder, "id",
				fieldLabel, true);
}

````

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by MarinhoFeliphe** [comment]
@p-albuquerque, nice pull request! :tada:

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:forward

**Pull request closed without merging**